### PR TITLE
Feat/quote service

### LINKED ISSUE #26: feat: rate modifiers and rules
For implementing this, we require:

- [x] windingtree/win-stays#99
- [x] #45
- [x] #46

### LINKED ISSUE #45: Implementation of rules logic (`SearchService`)

### LINKED ISSUE #46: Implementation of rate modifiers (`QuoteService`)

### PR DESCRIPTION
**Pull request by kostysh**
This PR adds:

- an implementation of the `QuoteService`
- tests

### CODE REVIEW

File: src/services/QuoteService.ts [hunk 1]
```diff
@@ -1,12 +1,241 @@
...
+    }
+
+    if (losSpaceModifier) {
+      const daysTotal = DateTime.fromObject(ask.checkOut).diff(
+        DateTime.fromObject(ask.checkIn)
+      ).days;
+      let applicable = false;
+
+      switch (losSpaceModifier.condition) {
+        case 0:
```
Comment (mfw78) [inline src/services/QuoteService.ts]: Consider using the `enum` from the `lpms.Condition`

File: src/services/QuoteService.ts [hunk 2]
```diff
@@ -1,12 +1,241 @@
...
+
+      if (applicable) {
+        switch (losSpaceModifier.valueOneof.oneofKind) {
+          case 'ratio':
+            total = total
+              .mul(BigNumber.from(losSpaceModifier.valueOneof.ratio.p))
+              .div(BigNumber.from(losSpaceModifier.valueOneof.ratio.q));
+            break;
+          case 'fixed':
+            total = BigNumber.from(losSpaceModifier.valueOneof.fixed);
```
Comment (mfw78) [inline src/services/QuoteService.ts]: `fixed` is to be interpreted as a "fixed" _modifier_, so this represents a constant either addition or subtraction. The value specified in the protobuf is an `int`, so by definition may be positive or negative. Therefore:

`total = total.add(BigNumber.from(losSpaceModifier.valueOneof.fixed));`

Comment (kostysh) [reply src/services/QuoteService.ts]: fiixed

File: src/services/QuoteService.ts (hunk 1 above)
Comment (kostysh) [reply src/services/QuoteService.ts]: fixed

**Review by mfw78 — APPROVED** [review]
LGTM!

**Comment by mfw78** [comment]
@kostysh Before merging, can you please address / confirm that the 'todo' in the PR description is complete?

**Comment by kostysh** [comment]
All points from TODO have been done.

**Pull request merged**